### ISSUE TITLE
Disallow `datetime.today()` in favor of `datetime.now()`

### ISSUE THREAD

**Issue by GideonBear** [astral-sh/ruff#23392]
### Summary

(Written pretty authoritatively, just my opinion, makes sense to me. Feel free to challenge, or give a difference between `datetime.today()` and `datetime.now()`.)

[`datetime.today()` has the same behavior as `datetime.now()`](https://docs.python.org/3/library/datetime.html#datetime.datetime.today). I propose to disallow `datetime.today()` and to recommend (and unsafely replace it with) `datetime.now()`, for the following reasons:
* The user and/or reader might assume it has the same behavior as `datetime.combine(date.today(), datetime.min.time())`
* The user may have intended to write `date.today()`
* The name `today` is (IMO) incorrect, or at least confusing; if I ask someone "what day is it?" I don't expect to hear "18 Feb 2026 15:17:09". The name `now` is clearer.
* `now` has the possibility to add a `tz` argument
* AFAICT it's only there because `datetime` inherits from `date`

Clarification:
```python
>>> from datetime import datetime, date
>>> date.today()
datetime.date(2026, 2, 18)
>>> datetime.today()
datetime.datetime(2026, 2, 18, 15, 5, 43, 133921)
>>> datetime.now()
datetime.datetime(2026, 2, 18, 15, 5, 47, 289484)
>>> datetime.combine(date.today(), datetime.min.time())
datetime.datetime(2026, 2, 18, 0, 0)
```

**Comment by ntBre** [comment]
I think we have a rule for this: [call-datetime-today (DTZ002)](https://docs.astral.sh/ruff/rules/call-datetime-today/#call-datetime-today-dtz002). Unless you're looking for something slightly different?

**Comment by GideonBear** [comment]
I didn't know that existed, because I ignore `DTZ`... Sorry, didn't check in a different environment!

Now that I know this, I would still like a different (`RUF`) rule for this, since that rule only talks about the timezone advantages, and not what I listed. I assume that when I ignore `DTZ` I ignore all timezone-related rules, and nothing else.

(Given that you agree with my reasoning; e.g. would you accept this proposal if DTZ002 didn't exist yet?)
Ideally `--select DTZ` would trigger this, and `--select ALL --ignore DTZ` would also trigger this. I don't think there's a way to do that without two separate rules. The rules would be identical but serving completely different purposes, which I personally think is fine.

**Comment by ntBre** [comment]
No, I don't think it makes sense to have two identical rules with different motivations. You can ignore DTZ and still select this rule if you want it to be active.

We could consider adding additional information to the rule documentation, but I'm not sure the motivation here is broadly applicable.

**Comment by GideonBear** [comment]
> No, I don't think it makes sense to have two identical rules with different motivations.

Okay, then I don't think this is going to happen without better grouping being implemented. I don't even know if the planned grouping would give us a way to do this.

> We could consider adding additional information to the rule documentation

Adding information to the rule documentation would not help in discovery.

> but I'm not sure the motivation here is broadly applicable.

I think it is, which is why I raised this. Feel free to close if you disagree, because that was the main motivation.

**Comment by GideonBear** [comment]
> You can ignore DTZ and still select this rule if you want it to be active.

Thanks, was already doing that! Just trying to "upstream" it to improve the experience for other `--ignore DTZ` users and minimize my config.

**Comment by MichaReiser** [comment]
Extending the documentation does make sense to me to mention something along:

> The name today is (IMO) incorrect, or at least confusing; if I ask someone "what day is it?" I don't expect to hear "18 Feb 2026 15:17:09". The name now is clearer.
